### PR TITLE
chore(resourcehierarchy): disambiguate prj-c-logging and prj-c-billing-logs

### PR DESCRIPTION
**Pull request by eeaton**
This PR changes one project name and moves one resource to another project for logical grouping where the implementation differs from the original design.

**Change 1: improve unclear project names**

- `prj-c-logging` was intended to contain all the audit logs from across the org. 
- `prj-c-billing-logs` was intended to contain the export of detailed cost data to BigQuery. "logs" is a misleading name, because logs are not involved.  I proposed `prj-c-billing-export` as a more clear name for the purpose.

**Change 2: location of billing account log sink**

In v4 changes, we added a billing account log sink to collect admin activity logs related to changes on the Billing Account. This should be grouped into the same project as other admin activity logs from across the org, but was accidentally deployed to the project for cost data exports, so I've moved the sink destination to `prj-c-logging`

### CODE REVIEW

**Comment by eeaton** [comment]
@sleighton2022 can I get a review on this one? non-functional changes

File: 1-org/envs/shared/README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@
 | logs\_export\_storage\_bucket\_name | The storage bucket for destination of log exports |
 | network\_folder\_name | The network folder name. |
 | org\_audit\_logs\_project\_id | The org audit logs project ID. |
-| org\_billing\_logs\_project\_id | The org billing logs project ID |
+| org\_billing\_export\_project\_id | The org billing logs project ID |
```
Comment (daniel-cit) [inline 1-org/envs/shared/README.md:48]:
```suggestion
| org\_billing\_export\_project\_id | The org billing export project ID |
```

File: 1-org/envs/shared/outputs.tf [hunk 2]
```diff
@@ -49,8 +49,8 @@ output "org_audit_logs_project_id" {
   description = "The org audit logs project ID."
 }
 
-output "org_billing_logs_project_id" {
-  value       = module.org_billing_logs.project_id
+output "org_billing_export_project_id" {
+  value       = module.org_billing_export.project_id
   description = "The org billing logs project ID"
```
Comment (daniel-cit) [inline 1-org/envs/shared/outputs.tf:54]:
```suggestion
  description = "The org billing export project ID"
```

**Review by daniel-cit — APPROVED** [review]
just two minor changes

**Review by apeabody — APPROVED** [review]
approved by @daniel-cit

**Pull request merged**